### PR TITLE
[Merged by Bors] - feat: port yaml files from mathlib3

### PR DESCRIPTION
**Pull request by eric-wieser**
This was done with a locally-hacked version of `fix-comments.py`, then manually fixing the last 50 or so missing instance names.

This includes CI that verifies the names are valid.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/checkYaml.lean [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import Mathlib.Tactic.Basic
```
Comment (urkud) [inline scripts/checkYaml.lean:1]: Could you please add a copyright header and some docs to this file?

Comment (eric-wieser) [reply scripts/checkYaml.lean:1]: Yes, but not until CI finished. It need to rebuild all of mathlib because I touched the lakefile...

Comment (eric-wieser) [reply scripts/checkYaml.lean:7]: Done

File: docs/overview.yaml [hunk 2]
```diff
@@ -1,490 +1,490 @@
...
-    free algebra of a commutative ring: 'free_algebra'
-    tensor product of algebras: 'algebra.tensor_product.tensor_product.algebra'
-    tensor algebra of a commutative ring: 'tensor_algebra'
-    Lie algebra: 'lie_algebra'
-    exterior algebra: 'exterior_algebra'
-    Clifford algebra: 'clifford_algebra'
+    associative algebra over a commutative ring: 'Algebra'
+    the category of algebras over a ring: 'AlgebraCat'
+    free algebra of a commutative ring: 'FreeAlgebra'
+    tensor product of algebras: 'Algebra.TensorProduct.instAlgebraTensorProductToAddCommMonoidToNonUnitalNonAssocSemiringToNonAssocSemiringToAddCommMonoidToNonUnitalNonAssocSemiringToNonAssocSemiringToModuleToModuleInstSemiringTensorProductToAddCommMonoidToNonUnitalNonAssocSemiringToNonAssocSemiringToAddCommMonoidToNonUnitalNonAssocSemiringToNonAssocSemiringToModuleToModule'
```
Comment (eric-wieser) [inline docs/overview.yaml:100]: This name isn't exactly ideal!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**